processor/src/main/java/org/mapstruct/ap/internal/processor/creation/MappingResolverImpl.java

```diff
@@ -163,6 +166,39 @@ public MethodReference getFactoryMethod(final Method mappingMethod, Type targetT
             matchingFactoryMethod.getParameterBindings() );
     }
 
+    private MethodReference findBuilderFactoryMethod(Method mappingMethod, Type targetType) {
+        if ( targetType.getBuilderType() == null ) {
+            return null;
+        }
+        Type builderType = targetType.getBuilderType();
+
+        Type returnType = mappingMethod.getReturnType();
+        List<ExecutableElement> builderCreators = new ArrayList<ExecutableElement>();
+        for ( ExecutableElement executableElement : Executables.getAllEnclosedExecutableElements(
+            elementsUtils,
+            returnType.getTypeElement()
+        ) ) {
+            if ( !executableElement.getModifiers().containsAll( Arrays.asList( Modifier.PUBLIC, Modifier.STATIC ) )
+                || !executableElement.getParameters().isEmpty()
+                || !typeUtils.isSameType( executableElement.getReturnType(), builderType.getTypeMirror() )) {
+                continue;
```

Comment:
I would add the condition (stop boolean) to the for statement. Don't like `continue`..

Reply:
You mean use `builderCreators.add( executableElement );` when that thing in that is `true`?

Reply:
No .. more in the loop it self.. so: use a classic iterator and add a boolean condition.. Or.. Java 8 :smile:

Reply:
Java 8 would be awesome. But we said 1.3 would stay on the same. Let's do that for 2.0 :)